### ISSUE TITLE
Replacing NaN values by zeros seems to have some unwanted side effects

### ISSUE THREAD

**Issue by BryanGuillaume** [NISOx-BDI/SwE-toolbox#150]
Following the request to replace NaN values by zeros in the result images to prevent issues with other software (see this [commit](https://github.com/NISOx-BDI/SwE-toolbox/pull/145/commits/a144fd93e1681e88e88c086fcd609970803426f9)), it seems that there are some undesirable side effects. For example, the minimum number of degrees of freedom is now 0:

<img width="219" alt="Screenshot 2019-11-12 at 16 36 54" src="https://user-images.githubusercontent.com/4426961/68685596-a61fde00-056a-11ea-96c7-1b30a2294f97.png">

I suspect that this is due to the fact that the edf image is read entirely while it would be more appropriate to read only the in-mask data.  I will review all the possible side effects that this change could cause and try to fix them

**Comment by BryanGuillaume** [comment]
I have detected two other bugs related to this:

1. The computation of the 5% FDR threshold in `swe_list.m` is including the zeros values, which is likely to bias the threshold (see [here](https://github.com/NISOx-BDI/SwE-toolbox/blob/a48d5e5b708a2bc451cd7530e0d14be80cdeae83/swe_list.m#L450-L458))

2. The extraction of edf summary statistics (see [here](https://github.com/NISOx-BDI/SwE-toolbox/blob/a48d5e5b708a2bc451cd7530e0d14be80cdeae83/swe_list.m#L537-L547)) does not work anymore. Indeed, the line `edf = swe_data_read(xSwE.Vedf);` loads the data as a 3D array  and, before, the line `edf(isnan(edf)) = []` was vectorising it providing that there were some NaN values. Now, because all NaN values have been removed, there is no vectorisation of `edf` anymore. That causes the extraction of the max/min/median statistics by using, for example, `max(edf)` to fail as they now yield a matrix instead of a scalar, causing the toolbox to throw an error when extracting these summary statistics.

I will look if there are other issues that may occur.
